### DISCUSSION TITLE
Getting data from subscription (.On function)

### DISCUSSION THREAD

**Discussion by krzysztof-chwiolka** [supabase-community/supabase-csharp#62] [Q&A] [answered]
Hello, 

The documentation seems to not state any details on how exactly I should be taking the data out of the response payload for an .On function.

Replication is enabled in Supabase, then:

If I would to take the example from documentation and modify it to get some more data:
```
await supabase.From<City>().On(ChannelEventType.Update, (sender, args) =>
{
    Debug.WriteLine(args.Response.Payload.Data);
    // We get a response of what the type is
    Debug.WriteLine(args.Response.Payload.Data.Columns.Count);
    // Here the output is correct number of columns, but these are just the names of each column?
    
    // And lets say I want to grab from Record the new column I would do this I assume?:
    Debug.WriteLine(args.Response.Payload.Data.Record.Columns.Count);
    // Here the output is nothing.
});
```
All that I could get from all of the payload data that is passed, is that I was e.g. doing an update, gathering the column names of 'City' Table in this case. OldRecord and Record seem to have no data being passed in through?

What do I do to grab the changed data?

I guess I could work around this that some data has been pass through and use a .Get elsewhere, but I guess that's not the cleanest solution, any ideas?

**Comment by acupofjose** [comment, marked as answer]
On my phone at the moment, but you probably want: `args.Response.Model<T>()`! I need to clarify the docs.

**Reply by krzysztof-chwiolka** [reply]
Thank you for the suggestion but didn't get it to work proper, I tried:
```Debug.WriteLine(args.Response.Model<Cities>().TableName);```
also didn't get any response, should I be passing a non table model here?

Edit:
```Debug.WriteLine(args.Response.Model<Cities>());``` also outputs nothing

**Reply by acupofjose** [reply]
Okay, two things:

1. Make sure you have full replication identity on - see here: https://supabase.com/docs/reference/csharp/subscribe
2. The following works on my test setup:

```c#
 var reference = supabase.From<Models.Channel>();

await reference.On(ChannelEventType.All, (sender, ev) =>
{
  var model = ev.Response.Model<Models.Channel>();
  Debug.WriteLine($"{ev.Response.Event} - [{model.Id}]:{model.Slug}");
});
```

**Reply by krzysztof-chwiolka** [reply]
Thank you for your help, this technically worked but I had another issue actually. 
This was an incorrectly specified table in C# (had an int instead of string)...

Solved now, thank you for the rapid response :)

**Reply by acupofjose** [reply]
Glad to hear it!
